flume-ng-sources/flume-scribe-source/src/main/java/org/apache/flume/source/scribe/ScribeSourceConfiguration.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.flume.source.scribe;
+
+import org.apache.flume.Context;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.server.THsHaServer;
+import org.apache.thrift.server.TThreadPoolServer;
+import org.apache.thrift.server.TThreadedSelectorServer;
+import org.apache.thrift.transport.TFramedTransport;
+import org.apache.thrift.transport.TNonblockingServerTransport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ScribeSourceConfiguration {
+  private static final Logger LOG = LoggerFactory.getLogger(ScribeSourceConfiguration.class);
+
+  public enum ThriftServerType {
+    THSHA_SERVER("ThshaServer"),
+    TTHREADED_SELECTOR_SERVER("TThreadedSelectorServer"),
+    TTHREADPOOL_SERVER("TThreadPoolServer");
+
+    private String value;
+
+    ThriftServerType(String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
+    }
+
+    public static ThriftServerType getServerType(String value) throws IllegalArgumentException {
+      for(ThriftServerType v : values()) {
+        if (v.getValue().equalsIgnoreCase(value)) {
+          return v;
+        }
+      }
+      throw new IllegalArgumentException("Can find server type:" + value);
+    }
+  }
+
+  public static final int DEFAULT_PORT = 1499;
+  public static final int DEFAULT_WORKERS = 5;
+  public static final ThriftServerType DEFAULT_THRIFT_SERVER_TYPE = ThriftServerType.THSHA_SERVER;
+  public static final int DEFAULT_THREADED_SELECTOR_NUM_THREADS = 4;
+  public static final long DEFAULT_MAX_READ_BUFFER_BYTES = 16384000;
+  public static final int DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES = 16384000;
+  public static final String SCRIBE_CATEGORY = "category";
+
+
+  public int port;
+  public int workerThreadNum;
+  public int selectorThreadNum;
+  public ThriftServerType thriftServerType;
+  public long maxReadBufferBytes;
+  public int maxThriftFrameSizeBytes;
+
+  public void configure(Context context) {
+    port = context.getInteger("port", DEFAULT_PORT);
+    maxReadBufferBytes = context.getLong("maxReadBufferBytes", DEFAULT_MAX_READ_BUFFER_BYTES);
+    if (maxReadBufferBytes <= 0) {
+      maxReadBufferBytes = DEFAULT_MAX_READ_BUFFER_BYTES;
+    }
+
+    maxThriftFrameSizeBytes = context.getInteger("maxThriftFrameSizeBytes",
+        DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES);
+    if (maxThriftFrameSizeBytes <= 0) {
+      maxThriftFrameSizeBytes = DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES;
+    }
+
+    workerThreadNum = context.getInteger("workerThreads", DEFAULT_WORKERS);
+    if (workerThreadNum <= 0) {
+      workerThreadNum = DEFAULT_WORKERS;
+    }
+    try {
+      thriftServerType = ThriftServerType.getServerType(
+          context.getString("thriftServer", DEFAULT_THRIFT_SERVER_TYPE.getValue()));
+    } catch (IllegalArgumentException ex) {
+      LOG.error("Wrong thrift server type config", ex);
```

Comment:
I would include the bad value in the error message, to help finding the bad config.
Could we throw an exception here instead of the error logging?

Reply:
@majorendre  Hi, I updated the patch and added an IllegalArgumentException here. However, the caller which is ```  @Override
  public void configure(Context context) ``` can't throw any exception as it overrides an non-exception method.

Reply:
Hi @JohnZZGithub 
Thanks for the quick reply. 
I took a deeper look and I suggest not handling that  IllegalArgumentException at all. You can just let it propagate, Flume will handle it properly. I did test this, to be sure.
See my attached diff. I also added the Apache licence to avoid 'rat' errors.
As @szaboferee mentioned before, please add some test cases.
Regards,
Endre